src/main/kotlin/com/github/pool_party/pull_party_bot/commands/PingCommandHandlers.kt

```diff
@@ -207,29 +192,62 @@ val delete = newAdministratorCommand(
     }
 }
 
-/**
- * Delete all the parties of the chat.
- */
 val clear = newAdministratorCommand("clear", "shut down all the parties ever existed", HELP_CLEAR) { msg, _ ->
     val chatId = msg.chat.id
     clearCommandTransaction(chatId)
     sendMessage(chatId, ON_CLEAR_SUCCESS)
 }
 
-/**
- * Create a new party with given members.
- */
 val create = newCommand("create", "create new party", HELP_CREATE) { msg, args ->
     handlePartyChangeRequest(true, msg, args)
 }
 
-/**
- * Change an existing party.
- */
 val change = newCommand("change", "change an existing party", HELP_CHANGE) { msg, args ->
     handlePartyChangeRequest(false, msg, args)
 }
 
+val add = newCommand("add", "add new users to the given party", HELP_ADD) { msg, args ->
+    handlePartyChangeMembersRequest(msg, args, List<String>::plus)
+}
+
+val remove = newCommand("remove", "remove given users from the provided party", HELP_REMOVE) { msg, args ->
+    handlePartyChangeMembersRequest(msg, args) { newArgs, party ->
+        val partyName = newArgs[0]
+        val argSet = newArgs.subList(1, newArgs.size).asSequence().map { it.removePrefix("@") }.toSet()
+        listOf(partyName) + party.asSequence().map { it.removePrefix("@") }.filter { !argSet.contains(it) }
+    }
+}
+
+/**
+ * Handle both `add` and `remove` commands.
+ */
+private fun Bot.handlePartyChangeMembersRequest(
+    msg: Message,
+    args: String?,
+    concat: (List<String>, List<String>) -> List<String>
+) {
+    val chatId = msg.chat.id
+    val parsedArgs = parseArgs(args)?.distinct()
+
+    if (parsedArgs.isNullOrEmpty()) {
+        sendCaseMessage(chatId, ON_CHANGE_EMPTY, parseMode = "Markdown")
+        return
+    }
+
+    val partyList = partyCommandTransaction(chatId, parsedArgs[0])
+    if (partyList == null) {
+        sendCaseMessage(chatId, ON_CHANGE_REQUEST_FAIL, parseMode = "Markdown")
```

Comment:
Same issue with concreteness of message

Reply:
Same answer with concreteness of message